modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase

```diff
@@ -1911,6 +1911,48 @@ definition {
 		}
 	}
 
+	@description = "LPS-187081 - Verify the user cannot delete the only custom field in an object definition"
+	@priority = 5
+	test CannotDeleteSoleCustomField {
+		
+		task ("Given a custom object published with a custom field") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object",
+				objectName = "CustomObject",
+				pluralLabelName = "Custom Objects");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.addObjectFieldViaUI(
+				enableShowFiles = "true",
+				fieldAttachment = "Upload Directly from the User's Computer",
+				fieldLabel = "Custom Attachment",
+				fieldType = "Attachment");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject");
+		}
+
+		task ("When the user attempts to delete the custom field") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.selectKebabMenuOption(kebabOption = "Delete");
+		}
+
+		task ("Then a message is seen stating the custom field cannot be deleted and the custom field isn't deleted") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "Modal#BODY",
+				value1 = "\"Custom Attachment\" is the only field of the published object definition and cannot be deleted.");
+		}
```

Comment:
Like the text states, you need to assert the field itself isn't deleted. You need to include an assertion that the field is present on the page after asserting the text on the modal.